### PR TITLE
Adds missing unit test for select component

### LINKED ISSUE #376: Unit tests for Select component
## Problem
We have a custom Select menu (`/mathesar_ui/src/components/select`), which does not any tests yet.

## Proposed solution
We need to add unit tests to the Select menu.
Unit tests need to be placed within `/mathesar_ui/src/components/select/__meta__`. Refer button component.
Naming convention: `Select.test.ts`

## Update
* @HenryRocha did great work writing a number of tests in https://github.com/centerofci/mathesar/pull/812.
* We still require one more test as mentioned in https://github.com/centerofci/mathesar/issues/376#issuecomment-971468648.
* Anyone working on this issue would be writing that particular test, and can suggest and write additional tests that may not have been covered.

### PR DESCRIPTION
**Pull request by nimishbongale**
Fixes #376

**Technical details**
Adds the missing unit test for select component 'list scrolls automatically to display currently selected option'

**Screenshots**
Screenshot 2022-03-27 at 1 03 10 AM

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/component-library/select/__meta__/Select.test.ts [hunk 1]
```diff
@@ -180,3 +180,58 @@ test('hide select dropdown by pressing ESC', async () => {
...
+
+  // select the option.
+  await fireEvent.keyDown(selectBtn, { key: 'Enter', code: 'Enter' });
+  await fireEvent.keyUp(selectBtn, { key: 'Enter', code: 'Enter' });
+
+  // Reopen the dropdown
+  await fireEvent.click(selectBtn);
+
+  expect(optionElement.offsetTop).toBeGreaterThanOrEqual(
+    dropdownContainer ? dropdownContainer.scrollTop : 0,
```
Comment (pavish) [inline mathesar_ui/src/component-library/select/__meta__/Select.test.ts:228]: Obtaining the `dropdownContainer` element should be done before this expect statement. We also need to ensure that `dropdownContainer` is present in the dom.

This test would still pass if `dropdownContainer` is not even rendered, which should not be the case.

**Review by pavish — CHANGES_REQUESTED** [review]
@nimishbongale Thanks for the PR! I have a small comment, upon resolution, we can merge this PR in.

File: mathesar_ui/src/component-library/select/__meta__/Select.test.ts (hunk 1 above)
Comment (nimishbongale) [reply mathesar_ui/src/component-library/select/__meta__/Select.test.ts:227]: @pavish 

I've done the changes:

1. dropdownContainer now gets initialized right before the assertions
2. added a check to see if dropdownContainer is in the DOM

Comment (nimishbongale) [reply mathesar_ui/src/component-library/select/__meta__/Select.test.ts:227]: @pavish 

I've done the changes:

1. dropdownContainer now gets initialized right before the assertions
2. added a check to see if dropdownContainer is in the DOM

**Review by pavish — APPROVED** [review]
@nimishbongale Looks good to me!

**Pull request merged**